lib/pool.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const storage = Symbol('storage');
+
+class Pool {
+  constructor(factory) {
+    this.factory = factory;
+    this[storage] = [];
+  }
+
+  put(value) {
+    this[storage].push(value);
+  }
+
+  get() {
+    if (this[storage].length === 0) {
+      if (this.factory) {
+        return this.factory();
+      } else {
+        return null;
+      }
+    }
+    return this[storage].shift();
```

Comment:
I guess using `pop()` would be more efficient?

Reply:
But this will change behaviour - it will be stack instead of the query. Does it matter here?

Reply:
@SemenchenkoVitaliy not at all, I can even return a random element here. The single Pool instance is supposed to store multiple instances of the same type and the users of this class are not supposed to depend on the order in which the objects are returned from `get()` after being `put()` into the pool.